### PR TITLE
MFW-2853-Policy-Manager

### PR DESCRIPTION
**Pull request by AlistairJohnParker**
I expect there is more to be done here but merging it up for comment.

### CODE REVIEW

File: v1/policy_manager/policy_manager_schema.json [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+                    }
+                }
+            }
+        },
+        "policy_manager_categories": {
+            "type": "string",
+            "enum": [
+                "discovery",
+                "geoip",
+                "threadprevention",
```
Comment (rasmussen-untangle) [inline v1/policy_manager/policy_manager_schema.json:105]: "threatprevention"

File: v1/policy_manager/policy_manager_schema.json [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+                "webfilter"
+            ],
+            "discovery": {
+                "$ref": "file:discovery/discovery_schema.json#/definitions/discovery_settings"
+            },
+            "geoip": {
+                "$ref": "file:geoip/geoip_schema.json#/definitions/geoip_settings"
+            },
+            "threadprevention": {
```
Comment (rasmussen-untangle) [inline v1/policy_manager/policy_manager_schema.json:114]: "threatprevention"

File: v1/policy_manager/policy_manager_schema.json [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                    "type": "boolean"
+                },
+                "sensitivity": {
+                    "type": "integer"
+                },
+                "category": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/policy_manager_categories"
+                    }
```
Comment (rasmussen-untangle) [inline v1/policy_manager/policy_manager_schema.json:96]: redirect, sensitivity and category seems plugin specific.  Don't think that goes here?. @dhadarau @AlistairJohnParker  Would like to keep any "plugin" specific configuration in a settings structure specific to that plugin.

File: v1/policy_manager/test_settings.json [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+            }
+        ],
+        "policies": [
+            {
+                "configurations": [
+                    "ae25229b-c3d5-4e06-9da2-ceeaa9b1c50c",
+                    "d9b27e4a-2b8b-4500-a64a-51e7ee5777d5",
+                    "95922aef-3968-4de5-9385-3f426fb32b99"
+                ],
+                "defaults": false,
```
Comment (rasmussen-untangle) [inline v1/policy_manager/test_settings.json:192]: I don't think this is needed.? The default is the main configuration as is now. E.g. if users do not have any policies defined.

File: v1/policy_manager/policy_manager_schema.json [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    "$schema": "http://json-schema.org/draft-06/schema#",
+    "definitions": {
+        "policy_manager_settings": {
+            "type": "object",
+            "description": "Policy Manager settings",
+            "properties": {
+                "enabled": {
+                    "type": "boolean"
+                },
+                "flow": {
```
Comment (dhadarau) [inline v1/policy_manager/policy_manager_schema.json:11]: In tests is used `flows`. I suppose it should be `flows` or `flowCategories` as initially in the ticket description

File: v1/policy_manager/policy_manager_schema.json [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/policy_manager_flow"
+                    }
+                },
+                "condition": {
```
Comment (dhadarau) [inline v1/policy_manager/policy_manager_schema.json:17]: I think `condition` slipped here from flow conditions. I think it should be removed as we do not have a Policy `condition`. Also note the typo `polocy_manager_condition` from `$ref`

File: v1/policy_manager/policy_manager_schema.json [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+                    "items": {
+                        "$ref": "#/definitions/policy_manager_configuration"
+                    }
+                },
+                "policy": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/policy_manager_policy"
+                    }
+                }
```
Comment (dhadarau) [inline v1/policy_manager/policy_manager_schema.json:34]: Same for `configuration` and `policy` as for `flow`. We maybe want to use plurals or even the initial ticket desc: `policy` => `policyDefinitions` or `policies`, `configuration` => `policyConfigurations` or `configurations`.

It seems to me that in `test_settings.json` are used plurals.

File: v1/policy_manager/test_settings.json [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+                        }
+                    ]
+                }
+            },
+            {
+                "description": "test",
+                "id": "56bfd272-5bef-4a53-8a21-304ebf24bdee",
+                "name": "test ND",
+                "discovery": {
+                    "enabled": true,
```
[the remark this answers is not in the document]
Comment (AlistairJohnParker) [reply v1/policy_manager/test_settings.json:98]: Maybe it doesn't belong here.

File: v1/policy_manager/policy_manager_schema.json (hunk 3 above)
Comment (AlistairJohnParker) [reply v1/policy_manager/policy_manager_schema.json:101]: Ok... removed from this level.

File: v1/policy_manager/policy_manager_schema.json (hunk 6 above)
Comment (AlistairJohnParker) [reply v1/policy_manager/policy_manager_schema.json:17]: Agreed

File: v1/policy_manager/test_settings.json (hunk 4 above)
Comment (AlistairJohnParker) [reply v1/policy_manager/test_settings.json:192]: Removed

**Review by rasmussen-untangle — APPROVED** [review]

**Pull request merged**